packages/test/test-end-to-end-tests/src/test/handleValidation.spec.ts

```diff
@@ -0,0 +1,425 @@
+		// 			traits: {
+		// 				handle,
+		// 			},
+		// 		};
+		// 		treeRoot.applyEdit(
+		// 			Change.insertTree(
+		// 				handleNode,
+		// 				StablePlace.atStartOf({
+		// 					parent: treeRoot.currentView.root,
+		// 					label: legacyNodeId,
+		// 				}),
+		// 			),
+		// 		);
+
+		// 		const rootNode = treeRoot.currentView.getViewNode(treeRoot.currentView.root);
+		// 		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		// 		const nodeId = rootNode.traits.get(legacyNodeId)![0];
+		// 		const change: Change = Change.setPayload(nodeId, handle);
+		// 		treeRoot.applyEdit(change);
+		// 	},
+		// },
+		{
+			type: registerId,
+			fn: async (defaultDataStore, handle) => {
+				const registerRoot = await defaultDataStore.getSharedObject(registerId);
+				registerRoot.write("B", handle);
+			},
+		},
+		{
+			type: queueId,
+			fn: async (defaultDataStore, handle) => {
+				const queueRoot = await defaultDataStore.getSharedObject(queueId);
+				queueRoot.add(handle);
+			},
+		},
+		// {
+		// 	type: migrationShimId,
+		// 	fn: async (defaultDataStore, handle) => {
+		// 		const migrationShimRoot = await defaultDataStore.getSharedObject(migrationShimId);
+		// 		const tree = migrationShimRoot.currentTree as LegacySharedTree;
+		// 		const legacyNodeId: TraitLabel = "inventory" as TraitLabel;
+
+		// 		const handleNode: BuildNode = {
+		// 			definition: legacyNodeId,
+		// 			traits: {
+		// 				handle,
+		// 			},
+		// 		};
+		// 		tree.applyEdit(
+		// 			Change.insertTree(
+		// 				handleNode,
+		// 				StablePlace.atStartOf({
+		// 					parent: tree.currentView.root,
+		// 					label: legacyNodeId,
+		// 				}),
+		// 			),
+		// 		);
+
+		// 		const rootNode = tree.currentView.getViewNode(tree.currentView.root);
+		// 		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		// 		const nodeId = rootNode.traits.get(legacyNodeId)![0];
+		// 		const change = Change.setPayload(nodeId, handle);
+		// 		tree.applyEdit(change);
+		// 	},
+		// },
+	];
+
+	for (const storeHandle of handleFns) {
+		it(`store handle in dds: ${storeHandle.type}`, async () => {
+			const { container1, provider } = await setup();
+			const defaultDataStore = (await container1.getEntryPoint()) as ITestFluidObject;
+			const runtime = defaultDataStore.context.containerRuntime;
+
+			const dataStoreB = await runtime.createDataStore(["default"]);
+			const dataObjectB = (await dataStoreB.entryPoint.get()) as ITestFluidObject;
+			const idB = dataObjectB.context.id;
+
+			await storeHandle.fn(defaultDataStore, dataObjectB.handle);
+
+			await provider.ensureSynchronized();
+			const container2: IContainerExperimental =
+				await provider.loadTestContainer(testContainerConfig);
+			await waitForContainerConnection(container2);
+			const default2 = (await container2.getEntryPoint()) as ITestFluidObject;
+			const dds2 = await default2.getSharedObject(storeHandle.type);
+
+			let handleB;
+			switch (storeHandle.type) {
```

Comment:
rather than have a case here, can we add another function the array above to preform the read just like we do for store